From 90cd267debe67d7d6896514be818fb529d25e517 Mon Sep 17 00:00:00 2001
From: Marcelo Leitner <mleitner@redhat.com>
Date: Tue, 21 Oct 2014 16:19:03 -0400
Subject: [net] netfilter: nf_tables: indicate family when dumping set elements

Message-id: <20141021161903.22888.34585.stgit@localhost.localdomain>
Patchwork-id: 97969
O-Subject: [RHEL7 net PATCH 073/124] netfilter: nf_tables: indicate family when dumping set elements
Bugzilla: 1148041 1155088
RH-Acked-by: Florian Westphal <fwestpha@redhat.com>

Backport of upstream commit: 6403d96254c7c44fdfa163248b1198c714c65f6a

 commit 6403d96254c7c44fdfa163248b1198c714c65f6a
 Author: Pablo Neira Ayuso <pablo@netfilter.org>
 Date:   Wed Jun 11 19:05:28 2014 +0200

    netfilter: nf_tables: indicate family when dumping set elements

    Set the nfnetlink header that indicates the family of this element.

    Signed-off-by: Pablo Neira Ayuso <pablo@netfilter.org>

Signed-off-by: Marcelo Ricardo Leitner <mleitner@redhat.com>
Signed-off-by: Jiri Benc <jbenc@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/net/netfilter/nf_tables_api.c b/net/netfilter/nf_tables_api.c
index 39369ea..ab4566c 100644
--- a/net/netfilter/nf_tables_api.c
+++ b/net/netfilter/nf_tables_api.c
@@ -2850,7 +2850,7 @@ static int nf_tables_dump_set(struct sk_buff *skb, struct netlink_callback *cb)
   goto nla_put_failure;
 
  nfmsg = nlmsg_data(nlh);
- nfmsg->nfgen_family = NFPROTO_UNSPEC;
+ nfmsg->nfgen_family = ctx.afi->family;
  nfmsg->version      = NFNETLINK_V0;
  nfmsg->res_id       = 0;
 
-- 
1.7.1